src/Symfony/Component/WebLink/EventListener/AddLinkHeaderListener.php

```diff
@@ -9,26 +9,28 @@
  * file that was distributed with this source code.
  */
 
-namespace Symfony\Component\Asset\EventListener;
+namespace Symfony\Component\WebLink\EventListener;
 
-use Symfony\Component\Asset\Preload\PreloadManager;
-use Symfony\Component\Asset\Preload\PreloadManagerInterface;
+use Psr\Link\LinkProviderInterface;
 use Symfony\Component\EventDispatcher\EventSubscriberInterface;
 use Symfony\Component\HttpKernel\Event\FilterResponseEvent;
 use Symfony\Component\HttpKernel\KernelEvents;
+use Symfony\Component\WebLink\HttpHeaderSerializer;
 
 /**
- * Adds the preload Link HTTP header to the response.
+ * Adds the Link HTTP header to the response.
  *
  * @author Kévin Dunglas <dunglas@gmail.com>
+ *
+ * @final
  */
-class PreloadListener implements EventSubscriberInterface
+class AddLinkHeaderListener implements EventSubscriberInterface
 {
-    private $preloadManager;
+    private $serializer;
 
-    public function __construct(PreloadManagerInterface $preloadManager)
+    public function __construct()
     {
-        $this->preloadManager = $preloadManager;
+        $this->serializer = new HttpHeaderSerializer();
```

Comment:
It's indeed an internal detail of the listener.

But the HttpHeaderSerializer has can be used alone through the component. I don't think it makes sense (for now) to expose it as a service but it may be useful for library creators to serialize a list of Link instances to a valid HTTP header value (it's almost all this component do).

I would keep this as is (btw the serializer is final but not internal for the same reason).